include/deal.II/lac/petsc_solver.h

```diff
@@ -141,21 +148,53 @@ namespace PETScWrappers
     void
     initialize(const PreconditionBase &preconditioner);
 
+    /**
+     * Return the PETSc KSP object.
+     */
+    KSP
+    petsc_ksp();
+
+    /**
+     * Conversion operator to gain access to the underlying PETSc type. If you
+     * do this, you cut this class off some information it may need, so this
+     * conversion operator should only be used if you know what you do.
+     */
+    operator KSP() const;
```

Comment:
Why do you need a conversion operator and an access function? I think in other places we only add a conversion operator (Vectors), right?

Reply:
Vectors and matrices have both. One is const (operator) and the other is not.